### PR TITLE
clear snapshot buffer pointer after releasing

### LINKED ISSUE #15508: darktable segfault
### Describe the bug

I was playing with the _sigmoid_ module's new _primaries_ controls using an SDR sRGB colour wheel image. I made a snapshot and was comparing the effects; I think the crash happened when I clicked on the snapshot to turn on the comparison. I had been dragging the sliders around quite wildly, so I'm not exactly sure.

### Steps to reproduce

I have been unable to reproduce the crash.

### Expected behavior

_No response_

### Logfile | Screenshot | Screencast

Copying the dump from the console, as it has a bit of processing history at the beginning, maybe it is of some use.
[crashlog.txt](https://github.com/darktable-org/darktable/files/13195941/crashlog.txt)

### Commit

_No response_

### Where did you install darktable from?

darktable.org

### darktable version

4.5.0+1014~g5f25f79812

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu 23.04

### Describe your system?

X11, 64 GB, NVidia 1060/6 GB, nvidia driver v525.125.06.

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

NVidia 1060/6 GB, nvidia driver v525.125.06

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

Built using:
./build.sh --prefix /home/kofa/darktable-master --build-type Release --install -DCUSTOM_CFLAGS=ON -DCMAKE_CXX_FLAGS="-O3 -march=native" -DCMAKE_C_FLAGS="-O3 -march=native"
GCC 12.3.0

### PR DESCRIPTION
**Pull request by dterrahe**
hopefully fixes #15508

@kofa73 and @kurld please test

### CODE REVIEW

**Comment by TurboGit** [comment]
@dterrahe : Looks good, wondering if the:

```
  if(s->buf)
    dt_free_align(s->buf);
  s->buf = NULL;
```

Should not be put in `_clear_snapshot_entry()`?

**Comment by dterrahe** [comment]
Sure! Makes sense.

Before #15436 the surface was destroyed in clear_snapshots but set to NULL in clean_snapshot_entry.

Longer term, as I mentioned to @jenshannoschwalm, I think it would be worth looking into replacing this with a reference counted pointer into a global cache (after it has been extended to hold scale and location per item).

File: src/libs/snapshots.c [hunk 1]
```diff
@@ -479,8 +479,10 @@ static void _clear_snapshot_entry(dt_lib_snapshot_t *s)
 
   g_free(s->module);
   g_free(s->label);
+  dt_free_align(s->buf);
```
Comment (TurboGit) [inline src/libs/snapshots.c:482]: I'm pretty sure you still need `if(s->buf)` here as `_clear_snapshot_entry` is called to initialized the struct in `gui_init`.

Comment (dterrahe) [reply src/libs/snapshots.c:482]: `dt_free_align(NULL)` is a no-op, like the other frees.

Comment (TurboGit) [reply src/libs/snapshots.c:482]: Indeed, a wrong recollection on my side, sorry for the noise.

**Pull request merged**